psydac/api/tests/test_api_2d_compatible_spaces.py

```diff
@@ -86,95 +91,126 @@ def run_system_1_2d_dir(f0, sol, ncells, degree):
     return l2_error
 
 #==============================================================================
-def run_system_2_2d_dir(f1, f2,u1, u2, ncells, degree):
-    # ... abstract model
-    domain = Square()
+def run_stokes_2d_dir(domain, f, ue, pe, *, homogeneous, ncells, degree, scipy=False):
 
+    # ... abstract model
     V1 = VectorFunctionSpace('V1', domain, kind='H1')
     V2 = ScalarFunctionSpace('V2', domain, kind='L2')
     X  = ProductSpace(V1, V2)
 
-    x,y = domain.coordinates
-
-    F = element_of(V1, name='F')
-
-    u,v = [element_of(V1, name=i) for i in ['u', 'v']]
-    p,q = [element_of(V2, name=i) for i in ['p', 'q']]
+    u, v = elements_of(V1, names='u, v')
+    p, q = elements_of(V2, names='p, q')
 
+    x, y  = domain.coordinates
     int_0 = lambda expr: integral(domain , expr)
 
-    
-    a  = BilinearForm(((u,p),(v,q)), int_0(inner(grad(u),grad(v)) + div(u)*q - p*div(v)) )
-    l  = LinearForm((v,q), int_0(f1*v[0]+f2*v[1]+q))
-    
-    bc = EssentialBC(u, 0, domain.boundary)
-    equation = find([u,p], forall=[v,q], lhs=a((u,p),(v,q)), rhs=l(v,q), bc=bc)
+    a  = BilinearForm(((u, p), (v, q)), int_0(inner(grad(u), grad(v)) - div(u)*q - p*div(v)) )
+    l  = LinearForm((v, q), int_0(dot(f, v)))
```

Comment:
This should be right, as the strong form of the Stokes equations only has a forcing term on the momentum equation. I need to double check.

Reply:
I confirm that the formulas in the code look correct to me. Accordingly, I can only conclude that the previous version contained a bug. Or maybe this was an attempt at stabilizing the solution @saidctb?

Reply:
yes, it was for stabilizing the solution

Reply:
Interesting! Could you give me a reference so I can look into that?

Reply:
I mean for the Lagrange multiplier

Reply:
For the readability of the code it is indeed better to point to a reference where the method is described. Otherwise (eg if the method is very simple), an explanation should be given in comments.

Reply:
it's deleted by the changes of Yaman

Reply:
@campospinto I think I understand now. At some point @saidctb had an additional term for ensuring that the average of the pressure was zero, through the use of a Lagrange multiplier. But this was handled completely by hand, and we have removed it here as these are unit tests for our API. Instead, we are just removing the average of the pressure field from the final solution.